src/Storage/Notmuch.hs

```diff
@@ -78,24 +85,31 @@ mailFilepath m dbpath =
 setNotmuchMailTags
   :: (MonadError Error m, MonadIO m)
   => FilePath
+  -> [T.Text]
+  -> ([T.Text] -> NotmuchMail -> NotmuchMail)
   -> NotmuchMail
   -> m NotmuchMail
-setNotmuchMailTags dbpath m = do
-  nmtags <- toNotmuchTags (view mailTags m)
-  bracketT (Notmuch.databaseOpen dbpath) Notmuch.databaseDestroy (tagsToMessage nmtags (view mailId m))
-  pure m
+setNotmuchMailTags dbpath tgs op m = do
+  let m' = op tgs m
```

Comment:
Not very happy on this here... can you think of solving this in a better way?

I've tried to first apply tags in a pure function, pass a list of `NotmuchMail`s to the storage module and write it to the database. This works fine if I'm adding tags to a selected `NotmuchMail` but changes quickly when dealing with a thread. I need to do an IO operation to grab a vector of mails which spoils the entire idea.